aws/src/main/java/org/apache/iceberg/aws/glue/LakeFormationCredentialsProvider.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aws.glue;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import software.amazon.awssdk.auth.credentials.AwsCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials;
+import software.amazon.awssdk.services.lakeformation.LakeFormationClient;
+import software.amazon.awssdk.services.lakeformation.model.GetTemporaryGlueTableCredentialsRequest;
+import software.amazon.awssdk.services.lakeformation.model.GetTemporaryGlueTableCredentialsResponse;
+
+public class LakeFormationCredentialsProvider implements AwsCredentialsProvider {
+  private LakeFormationClient lakeFormationClient;
+
+  private String tableArn;
+
+  public void initialize(Map<String, String> properties) {
+  }
+
+  @Override
+  public AwsCredentials resolveCredentials() {
+    GetTemporaryGlueTableCredentialsRequest getTemporaryGlueTableCredentialsRequest =
+            GetTemporaryGlueTableCredentialsRequest.builder()
+                    .tableArn(tableArn)
+                    .permissionsWithStrings(getPrivilegeList())
+                    .build();
+    GetTemporaryGlueTableCredentialsResponse response
+            = lakeFormationClient.getTemporaryGlueTableCredentials(getTemporaryGlueTableCredentialsRequest);
+    return AwsSessionCredentials.create(response.accessKeyId(), response.secretAccessKey(), response.sessionToken());
+  }
+
+  private List<String> getPrivilegeList() {
```

Comment:
this looks like test code?

Reply:
yes, I add this simple `resolveCredentials` to reduce complexity, will add cache later.